modules/packages/ZMQ.chpl

```diff
@@ -802,29 +798,33 @@ module ZMQ {
           if errno == EAGAIN then
             chpl_task_yield();
           else {
-            var errmsg = zmq_strerror(errno):string;
-            try! halt("Error in Socket.send(%s): %s\n".format(T:string, errmsg));
+            try throw_socket_error("Error in Socket.send(%s): %s\n");
           }
         }
       }
     }
 
     // send, enumerated types
     pragma "no doc"
-    proc send(data: ?T, flags: int = 0) where isEnumType(T) {
-      send(data:int, flags);
+    proc send(data: ?T, flags: int = 0) throws where isEnumType(T) {
+      try send(data:int, flags);
     }
 
     // send, records (of other supported things)
     pragma "no doc"
-    proc send(data: ?T, flags: int = 0) where (isRecordType(T) &&
-                                               (!isString(T))) {
+    proc send(data: ?T, flags: int = 0) throws where (isRecordType(T) &&
+                                                     (!isString(T))) {
       on classRef.home {
         var copy = data;
         param N = numFields(T);
-        for param i in 1..(N-1) do
-          send(getField(copy,i), ZMQ_SNDMORE | flags);
-        send(getField(copy,N), flags);
+        try {
+          for param i in 1..(N-1) do
+            send(getField(copy,i), ZMQ_SNDMORE | flags);
+        } catch e: TaskErrors {
```

Comment:
Why are we catching TaskErrors here? Where's the tasks?

Reply:
I must've had a stroke or something, I thought that was a `forall`. My bad.